From f6d34323d05ae73bf2404a5e50615f766c2bc01d Mon Sep 17 00:00:00 2001
From: Virtroid build <build@virtroid.invalid>
Date: Mon, 26 Oct 2026 23:23:40 +0000
Subject: [PATCH] Let a device turn on themed icons by default

The icon theme preference defaults to the monochrome theme when config_default_themed_icons is true, false (as before) unless a device overlays it. A theme the user has chosen, or turned off, is kept. Virtroid's runtimes set true.

Change-Id: I12bc0427e4a0a642009ec2c5acae1837e5b22e1c
---
 res/values/config.xml                                  |  3 +++
 .../launcher3/graphics/theme/ThemePreference.kt        | 10 +++++++++-
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/res/values/config.xml b/res/values/config.xml
index cb6e50b..8ff056f 100644
--- a/res/values/config.xml
+++ b/res/values/config.xml
@@ -17,6 +17,9 @@
     <!-- Miscellaneous -->
     <bool name="config_largeHeap">false</bool>
 
+    <!-- Whether themed (monochrome) icons are on until the user chooses. -->
+    <bool name="config_default_themed_icons">false</bool>
+
     <!-- A string pointer to the original app name string. This allows derived projects to
      easily override the app name without providing all translations -->
     <string name="derived_app_name" translatable="false">@string/lineageos_app_name</string>
diff --git a/src/com/android/launcher3/graphics/theme/ThemePreference.kt b/src/com/android/launcher3/graphics/theme/ThemePreference.kt
index 3f8f102..fe8010c 100644
--- a/src/com/android/launcher3/graphics/theme/ThemePreference.kt
+++ b/src/com/android/launcher3/graphics/theme/ThemePreference.kt
@@ -20,6 +20,7 @@ import androidx.annotation.VisibleForTesting
 import com.android.launcher3.ConstantItem
 import com.android.launcher3.LauncherPrefs
 import com.android.launcher3.LauncherPrefs.Companion.backedUpItem
+import com.android.launcher3.R
 import com.android.launcher3.dagger.LauncherAppSingleton
 import com.android.launcher3.graphics.theme.MonoIconThemeFactory.MONO_FACTORY_ID
 import com.android.launcher3.graphics.theme.MonoIconThemeFactory.MONO_THEME_CONTROLLER
@@ -99,7 +100,14 @@ private constructor(
         @JvmField val MONO_THEME_VALUE = ThemeValue(MONO_FACTORY_ID, MONO_THEME_CONTROLLER.themeID)
 
         private const val KEY_ICON_THEME = "icon_theme_id"
-        private val THEME_ID = backedUpItem(KEY_ICON_THEME, "")
+        private val THEME_ID =
+            backedUpItem(KEY_ICON_THEME, String::class.java) {
+                if (it.resources.getBoolean(R.bool.config_default_themed_icons)) {
+                    MONO_THEME_VALUE.toString()
+                } else {
+                    ""
+                }
+            }
 
         @VisibleForTesting val LEGACY_MONO_THEME_ICON = backedUpItem("themed_icons", false)
 
-- 
2.43.0

